docs/resources/virtual_environment_vm.md

```diff
@@ -108,355 +108,355 @@ output "ubuntu_vm_public_key" {
-    - `device` - (Required) The PCI device name for Proxmox, in form
+  - `device` - (Required) The PCI device name for Proxmox, in form
       of `hostpciX` where `X` is a sequential number from 0 to 3.
-    - `id` - (Optional) The PCI device ID. This parameter is not compatible
+  - `id` - (Optional) The PCI device ID. This parameter is not compatible
       with `api_token` and requires the root `username` and `password`
       configured in the proxmox provider. Use either this or `mapping`.
-    - `mapping` - (Optional) The resource mapping name of the device, for
+  - `mapping` - (Optional) The resource mapping name of the device, for
       example gpu. Use either this or `id`.
-    - `mdev` - (Optional) The mediated device ID to use.
-    - `pcie` - (Optional) Tells Proxmox to use a PCIe or PCI port. Some
+  - `mdev` - (Optional) The mediated device ID to use.
+  - `pcie` - (Optional) Tells Proxmox to use a PCIe or PCI port. Some
       guests/device combination require PCIe rather than PCI. PCIe is only
       available for q35 machine types.
-    - `rombar` - (Optional) Makes the firmware ROM visible for the VM (defaults
+  - `rombar` - (Optional) Makes the firmware ROM visible for the VM (defaults
       to `true`).
-    - `rom_file` - (Optional) A path to a ROM file for the device to use. This
+  - `rom_file` - (Optional) A path to a ROM file for the device to use. This
       is a relative path under `/usr/share/kvm/`.
-    - `xvga` - (Optional) Marks the PCI(e) device as the primary GPU of the VM.
+  - `xvga` - (Optional) Marks the PCI(e) device as the primary GPU of the VM.
       With this enabled the `vga` configuration argument will be ignored.
 - `usb` - (Optional) A host USB device mapping (multiple blocks supported).
-    - `host` - (Optional) The USB device ID. Use either this or `mapping`.
-    - `mapping` - (Optional) The resource mapping name of the device, for
+  - `host` - (Optional) The USB device ID. Use either this or `mapping`.
+  - `mapping` - (Optional) The resource mapping name of the device, for
       example usbdevice. Use either this or `id`.
-    - `usb3` - (Optional) Makes the USB device a USB3 device for the VM (defaults
+  - `usb3` - (Optional) Makes the USB device a USB3 device for the VM (defaults
       to `false`).
 - `initialization` - (Optional) The cloud-init configuration.
-    - `datastore_id` - (Optional) The identifier for the datastore to create the
+  - `datastore_id` - (Optional) The identifier for the datastore to create the
       cloud-init disk in (defaults to `local-lvm`).
-    - `interface` - (Optional) The hardware interface to connect the cloud-init
+  - `interface` - (Optional) The hardware interface to connect the cloud-init
       image to. Must be one of `ide0..3`, `sata0..5`, `scsi0..30`. Will be
       detected if the setting is missing but a cloud-init image is present,
       otherwise defaults to `ide2`.
```

Comment:
Please add note that `ideX` does not works with uefi/ovmf and `scsiX` is required. More people are starting to try cloud-init with uefi run into the issue (most recent [here](https://github.com/bpg/terraform-provider-proxmox/issues/736#issuecomment-1823218832))

Reply:
`ideX` seems to be working in [some cases](https://github.com/bpg/terraform-provider-proxmox/issues/586#issuecomment-1826288326).

I'm planning to add a separate section dedicated to cloud-init with more examples, will explore interfaces and limitations there.